src/CarbonAware.LocationSources/src/Configuration/LocationDataSource.cs

```diff
@@ -0,0 +1,56 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.LocationSources.Configuration;
+
+public class LocationDataSource
+{
+
+    private const string DefaultAzureLocationDataFile = "azure-regions.json";
+    private const string BaseDirectory = "location-sources/json";
+    private const string DirectoryRegExPattern = @"^[-\\/a-zA-Z_\d ]*$";
+
+    private string assemblyDirectory;
+    private string? dataFileLocation;
+
+    /// <summary>
+    /// Location data file location
+    /// </summary>
+    public string DataFileLocation
+    {
+        get => dataFileLocation!;
+        set
+        {
+            if (!IsValidDirPath(value))
+            {
+                throw new ArgumentException($"File path '{value}' contains not supported characters.");
+            }
+            dataFileLocation = Path.Combine(assemblyDirectory, BaseDirectory, value);
+        }
+    }
+
+    public string? Prefix { get ; set; }
+    public char? Delimiter { get;  set; }
```

Comment:
prefer non-nullable and giving users the ability to use multi-char delimiters like double underscore

```suggestion
    public string Delimiter { get;  set; } = string.Empty;
```